content/spin/serverless-ai-api-guide.md

```diff
@@ -0,0 +1,157 @@
+title = "Serverless API"
+template = "spin_main"
+date = "2023-09-05T09:00:00Z"
+enable_shortcodes = true
+[extra]
+url = "https://github.com/fermyon/developer/blob/main/content/spin/serverless-ai-api-guide.md"
+
+---
+- [Using Fermyon Serverless AI From Applications](#using-fermyon-serverless-ai-from-applications)
+
+Spin provides an interface for you to perform AI inference within Spin applications. By default, a given component of an app will not have access to any Serverless AI models. Access must be provided explicitly via the component manifest.  For example, a component could be given access to the llama2-chat model using:
+
+```toml
+[[component]]
+ai_models = ["llama2-chat"]
+```
+
+> Note: To deploy a Spin application (that performs AI inferencing) to Fermyon Cloud please visit the [Serverless AI](https://developer.fermyon.com/cloud/serverless-ai#accessing-private-beta) section in the documentation. It covers signing up for the private beta and setting up your Fermyon Serverless AI.
+
+## Using Fermyon Serverless AI From Applications
+
+The Spin SDK surfaces the Serverless AI interface to your language. The set of operations is common across all supporting SDKs:
+
+| Operation  | Parameters | Returns | Behavior |
+|------------|------------|---------|----------|
+| `infer`  | model `string`, prompt `string`, params `list` | `string`  | The `infer` is performed on a specific model. The name of the model is the first parameter provided (i.e. `llama2-v70b-chat`, `llama2-v13b-chat`, `llama2-v7b-chat` or other; passed in as a `string`). The second parameter is a prompt; passed in as a `string`. The third parameter is a mix of float and unsigned integers relating to inferencing parameters in this order: `max-tokens` (unsigned 32 integer), `repeat-penalty` (float 32), `repeat-penalty-last-n-token-count` (unsigned 32 integer), `temperature` (float 32), `top-k` (unsigned 32 integer) and `top-p` (float 32). The result from `infer` is a `string` |
+
+The exact detail of calling these operations from your application depends on your language:
+
+{{ tabs "sdk-type" }}
+
+{{ startTab "Rust"}}
+
+To use Serverless AI functions, the `llm` module from the Spin SDK provides the method: `infer`. The following snippet is from the [Rust sentiment analysis example](https://github.com/fermyon/ai-examples/sentiment-analysis-rust): 
+
+```rust
+use spin_sdk::{llm::{infer, InferencingModel::Llama2Chat}};
+
+// -- snip --
+
+let inferencing_result = infer(Llama2Chat, &PROMPT.clone().replace("<SENTENCE>", &request.sentence));
+```
+
+**General Notes**
+
+`infer` operation:
+
+- It takes in the following arguments - model name, prompt and ... 
+- The model name is ...
+- The optional third parameter which is an interface allows you to specify parameters such as `maxTokens`, `repeatPenalty`, `repeatPenaltyLastNTokenCount`, `temperature`, `topK`, `topP`.  
```

Comment:
note: the casing is off here. For example, `repeatPenaltyLastNTokenCount` should be `repeat_penalty_last_n_token_count`